src/inmanta/parser/plyInmantaParser.py

```diff
@@ -731,40 +733,52 @@ def p_constant_rstring(p: YaccProduction) -> None:
 
 def p_constant_mls(p: YaccProduction) -> None:
     "constant : mls"
-    p[0] = get_string_ast_node(p[1], p[1].location)
+    p[0] = get_string_ast_node(p[1], True)
     attach_from_string(p)
 
 
 format_regex = r"""({{\s*([\.A-Za-z0-9_-]+)\s*}})"""
 format_regex_compiled = re.compile(format_regex, re.MULTILINE | re.DOTALL)
 
 
-def get_string_ast_node(string: LocatableString, location: Location) -> Union[Literal, StringFormat]:
-    match_obj = format_regex_compiled.findall(str(string))
-    if len(match_obj) == 0:
+def get_string_ast_node(string: LocatableString, mls: boolean) -> Union[Literal, StringFormat]:
+    matches = [[str(string)[m.start() : m.end()], m.start(), m.end()] for m in format_regex_compiled.finditer(str(string))]
```

Comment:
@sanderr  I think the match could be on multiple lines like this:
```
"""{

x.n


}"""
```

Reply:
I mean if you use `match.group(1)`, this will always be single-line because the group does not allow any whitespace characters.

Reply:
I think I need the other lines later on to compute the location of x and n. To compute the line_offset based on the number of white lines and the start line. I don't see how I could do it otherwise?